### PR TITLE
Surface forces (after rebasing #488)

### PR DESCRIPTION
**Pull request by ilfreddy**
Taken from PR #488 from @moritzgubler and rebased with the AZORA functionality which has just been merged in master.

Rest of the description verbatim from #488 

This is my implementation of the idea I had of [calculating forces with surface integrals](https://arxiv.org/abs/2410.02299).

It works quite well and the forces are more accurate. Also, mrchem struggles to compute forces when the world precision is 1e-7 or smaller. Then it requires a lot of memory (more than 32 GB for a H2 molecule). This is not the case with my approach . Geometry optimizations seem to be reliable if the stopping criterion is chosen 10 * world_prec.

I have already put all the parameters into the parser, so it is quite simple to test and run my code (look for the section "- name: Forces" in the template.yaml input parser file).

It works with lda and gga functionals and for both closed and open shell systems.
At the moment, there is one thing that might be improved in the future:

The exact exchange stress is not computed. Therefore forces can not be computed for exact exchange. It shouldn't be too complicated to implement, but in a quick search I did not find a reference that derives it.

### CODE REVIEW

**Comment by ilfreddy** [comment]
@moritzgubler I think I have fixed what you asked. Could you check and the fixes, and then review the whole PR? Thereafter we can most likely merge it into master (assuming no test fails this time).

**Comment by ilfreddy** [comment]
@moritzgubler the test fails with the new default, so I explicitly used 0.6 in the test. Although the gradient should in principle be the same, I assume this is a low prec calculation with an SCF which is not converged either. I therefore assume it is not granted the computed gradient makes physical sense at this stage.

**Comment by ilfreddy** [comment]
If you are OK with my fixes and you can review the whole PR, I believe this gan go now, unless @stigrj wants to have a look first.

**Review by robertodr — APPROVED** [review]
Thanks @moritzgubler!

File: doc/users/user_ref.rst [hunk 1]
```diff
@@ -458,6 +458,33 @@ User input reference
...
+    **Default** ``medium``
+  
+    **Predicates**
+      - ``value.lower() in ['low', 'medium', 'high']``
+  
+   :radius_factor: Sets the radius of the surface used in the computation of forces. The radius is given by this factor times the distance to the neariest neighbour. Must be between 0.1 and 0.9. This should rarely need to be changed. Different values can change the accuracy of the forces. 
+  
+    **Type** ``float``
+  
+    **Default** ``0.6``
```
Comment (moritzgubler) [inline doc/users/user_ref.rst:486]: Didn't you change that to 0.5?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -36,17 +36,17 @@ add_subdirectory(properties)
...
 target_compile_definitions(mrchem PRIVATE 
     HIRSHFELD_SOURCE_DIR="${HIRSHFELD_SOURCE_DIR}"
     HIRSHFELD_INSTALL_DIR="${HIRSHFELD_INSTALL_DIR}"
-)
-
-target_compile_definitions(mrchem PRIVATE 
     AZORA_POTENTIALS_SOURCE_DIR="${AZORA_POTENTIALS_SOURCE_DIR}"
     AZORA_POTENTIALS_INSTALL_DIR="${AZORA_POTENTIALS_INSTALL_DIR}"
+    LEBVEDEV_SOURCE_DIR="${LEBVEDEV_SOURCE_DIR}"
+    LEBVEDEV_INSTALL_DIR="${LEBVEDEV_INSTALL_DIR}"
```
Comment (moritzgubler) [inline src/CMakeLists.txt:49]: I think you can delete these lines

File: src/driver.cpp [hunk 3]
```diff
@@ -57,6 +57,7 @@
 #include "qmoperators/one_electron/NuclearOperator.h"
 #include "qmoperators/one_electron/ZoraOperator.h"
 #include "qmoperators/one_electron/AZoraPotential.h"
+#include "qmoperators/one_electron/NablaOperator.h"
```
Comment (moritzgubler) [inline src/driver.cpp:60]: this is not needed

File: src/driver.cpp [hunk 4]
```diff
@@ -83,8 +84,10 @@
 #include "environment/LPBESolver.h"
 #include "environment/PBESolver.h"
 #include "environment/Permittivity.h"
-
+#include "surface_forces/SurfaceForce.h"
 #include "properties/hirshfeld/HirshfeldPartition.h"
+#include <fstream>
+#include <filesystem>
```
Comment (moritzgubler) [inline src/driver.cpp:90]: this is not needed

File: tests/hf_grad_blyp_surface_force/hf.inp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+$coords
+F           -1.194868474027    -0.244625419775     0.340936636348
+H           -1.672505086978    -0.395082674713     1.107306807211
+$end
+translate = false
+}
+
+Forces {
+    method = 'surface_integrals'
+    radius_factor = 0.6
```
Comment (moritzgubler) [inline tests/hf_grad_blyp_surface_force/hf.inp:19]: when the radius_factor is set here, the test should still work when the default is changed to 0.5

Comment (ilfreddy) [reply tests/hf_grad_blyp_surface_force/hf.inp:19]: Indeed it does. With the added bonus that we test setting a non-default value 😄

File: python/template.yml [hunk 6]
```diff
@@ -416,6 +416,34 @@ sections:
...
+        predicates:
+          - value.lower() in ['low', 'medium', 'high']
+        docstring: |
+          Precision of the surface integrals used in the computation of forces. Determines the number of Lebedev grid points used in the
+          surface integration.
+      - name: radius_factor
+        type: float
+        default: 0.6
+        docstring: |
+          Sets the radius of the surface used in the computation of forces.
```
Comment (moritzgubler) [inline python/template.yml:444]: Here, the default is also not 0.5 did you run the parameter update script?

**Comment by ilfreddy** [comment]
If @moritzgubler and @stigrj are happy this one can finally go for my part.

File: python/template.yml (hunk 6 above)
Comment (ilfreddy) [reply python/template.yml]: I fixed that too now :-) Thanks for pointing that out!

**Pull request merged**